src/jit/lsra.cpp

```diff
@@ -5680,16 +5679,14 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
-        if (intervalToUnassign != nullptr)
+        intervalToUnassign = availablePhysRegInterval->assignedInterval;
+        if (isAssigned(availablePhysRegInterval ARM_ARG(currentInterval->registerType)))
         {
-            RegRecord* physRegToUnassign = availablePhysRegInterval;
 #ifdef _TARGET_ARM_
-            // We should unassign a double register if availablePhysRegInterval is part of the double register
-            if (availablePhysRegInterval->assignedInterval->registerType == TYP_DOUBLE &&
-                !genIsValidDoubleReg(availablePhysRegInterval->regNum))
-                physRegToUnassign = findAnotherHalfRegRec(availablePhysRegInterval);
+            unassignPhysReg(availablePhysRegInterval, currentInterval->registerType);
+#else
+            unassignPhysReg(availablePhysRegInterval, intervalToUnassign->recentRefPosition);
```

Comment:
I still don't understand why these can't be merged, with an `ARM_ONLY` 2nd argument to the overload above. If `intervalToUnassign` is non-null, it *must* be `availablePhysRegInterval->assignedInterval` which must simply be spilled at its `recentRefPosition`. However, at this point this is something we could clean up later.